docs/html/development/getting-started.rst

```diff
@@ -8,7 +8,55 @@ This document is meant to get you setup to work on pip and to act as a guide and
 reference to the the development setup. If you face any issues during this
 process, please `open an issue`_ about it on the issue tracker.
 
-Development tools
+Running Developent Version
+==========================
+
+Just run `python src/pip` from cloned pip repo (https://github.com/pypa/pip):
+
+.. code-block:: console
+
+    $ python src/pip
+    
+That's it!
+
+You may want to skip to next chapter to read on how to run tests, or read on
+for details on how it works.
+
+Running `src/pip` will load all `pip` modules and
+`dependencies <https://github.com/pypa/pip/tree/master/src/pip/_vendor)>`_
+from the clone directory. This way they do not need to be installed on your
+system. The process of bundling dependencies with a project sources is called
+`vendoring`, but because there is no PEP for that, other Python won't work
+this way and still require `pip` for installation.
+
+If you want to practice `virtualenv` development, then `pip` inside can be
+relinked directly to development sources by installing it in
```

Comment:
This is very confusingly worded. If you're just saying "this is how you use virtualenv for development" then fine, but it doesn't read like that. And anyway, using virtualenv and editable installs for development isn't a workflow that's specifically for pip, so while this section might be helpful for the odd person who doesn't know how to do it, it's not really needed in general.

Also, this is much more reliable than your proposed "just run pip" approach, so I'd rather see this as the default approach, with your suggestion only recommended for people who (for whatever reason - I can't honestly think of any reasonable ones) don't want to use a virtualenv.

Reply:
I didn't like the wording too, but I want to explain the how `--editable` option works. The way it is explained by `pip` is by vague reference to `setuptools` documentation:

    -e, --editable <path/url>   Install a project in editable mode (i.e. setuptools "develop mode") from a local
                              project path or a VCS url.

Which look like undocumented - you won't find find this mode in setuptools docs, which refer to `easy_install` docs, where no "develop mode" either - https://setuptools.readthedocs.io/en/latest/search.html?q=%22develop+mode%22 The problem is that you're not only exposing people to `virtualenv`, but also to decades of Python packaging problem with keyword like `setuptools` and `easy_install` that `pip` and `wheels` are trying to solve. It will take days to learn and [it is not for Millenials](https://www.youtube.com/watch?v=hER0Qp6QJNU) who need immediate gratification to keep them on track (speaking of myself). I mean you guys is the only reason I get back to this. )

Last time I tried to use `virtualenv` I ended up setting up `virtualenvwrapper`, and I tell you - there is some big reason why so many `virtualenv` forks/alternatives exists. So before I edit this text, maybe I can hear arguments why would anyone recommend `virtualenv` for `pip` development? The question is two fold:

1. Why recommend a tool like `virtualenv` at all if pip just works out of the box?
2. Why recommend old `virtualenv` and not more modern `virtualenvwrapper` or alternatives?

When you say `this is much more reliable` - it doesn't sound like an argument to me. Why it is much more reliable? Let's make technical doc technical. It is development doc asking for the whole gory details after all. My argument that you don't need to be exposed to all this complexity, because there is no way you can mess with your system dependencies while developing `pip`. You may use `virtualenv` for test run `pip install`, but that's another story - "How to create virtualenv with custom `pip`?" where you probably want to keep separate envs for every pip modification to be able to compare state and debug later.

Sorry, I am in a rush again. Can't smile. :)